packages/pages-shared/src/asset-server/handler.ts

```diff
@@ -34,6 +34,8 @@ export const HEADERS_VERSION = 2;
 export const HEADERS_VERSION_V1 = 1;
 export const ANALYTICS_VERSION = 1;
 
+const ALLOW_LISTED_EARLY_HINT_LINK_ATTRIBUTES = ["rel", "as", "href"];
```

Comment:
what is meant by `LINK` here? The HTTP `Link` entity-header field https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Link?

also if the above tries to say that we allow Early Hints Links to have only those 3 attributes, maybe call that const `ALLOWED_EARLY_HINT_LINK_ATTRIBUTES`?

and last question. According to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Link#parameters the `Link` header supports the same params as a regular `<link>` element. Is there a specific reason why we're excluding all other attrs?

Reply:
1. No, it's the `<link>` element attributes here, since that's what we are trying to parse out and convert into the `Link` headers. If a `<link>` element has anything other than these attributes, it is ignored.
2. Yeah, `ALLOWED_*` is probably better choice, I'll make that change.
3. I'm not sure that's entirely accurate. It might be, but until we confirm each attribute with every popular browser, I want to remove them at risk of accidentally prioritizing incorrectly.

Reply:
My bad. I thought the mapping was the other way around.

2. ✅ 

3. Got it. I couldn't find anything about browser compat either unfortunately. I'd say in this case pls add the above as a comment (above line 37 perhaps?) to help us understand why we took this decision. Do we want to add a backlog task to remind ourselves to check on browser compat status later on?

Reply:
3. Will add comment. We'll listen for feedback from users and see how important these other attributes are before scheduling any work around it :)

Reply:
> Is there a specific reason why we're excluding all other attrs?

`blocking`, `imagesrcset`, `imagesizes` and `media` are only valid in the DOM & only `as`, `crossorigin`, `integrity` & `type` are used in Early Hints processing.

`crossorigin` is mainly for `font` preloading I believe - might have other uses though.

> I couldn't find anything about browser compat either unfortunately

I'm not aware of any other browser than Chrome 103+ supporting it - of which, only supports the following `as` values.

```c
// Represents subset of possible values for `as` attribute of the Link header.
enum LinkAsAttribute {
  kUnspecified,
  kFont,
  kImage,
  kScript,
  kStyleSheet,
};
```

https://html.spec.whatwg.org/#early-hints